src/rotator_library/providers/anthropic_auth_base.py

```
@@ -0,0 +1,572 @@
+        Compatible with the proxy's startup credential processing flow.
+        """
+        path = creds_or_path if isinstance(creds_or_path, str) else None
+
+        if isinstance(creds_or_path, dict):
+            display_name = creds_or_path.get("_proxy_metadata", {}).get(
+                "display_name", "in-memory object"
+            )
+        else:
+            display_name = Path(path).name if path else "in-memory object"
+
+        lib_logger.debug(f"Initializing Anthropic token for '{display_name}'...")
```

Ruff (0.15.4)

[warning] Logging statement uses f-string

(G004)